### ISSUE TITLE
Rule Change: add `skipMath` and `skipInlineMath` options to `no-tab`

### ISSUE THREAD

**Issue by lumirlumir** [eslint-markdown/eslint-markdown#685]
## Overview

Add the **`skipMath` and `skipInlineMath` options** to the [no-tab](https://github.com/lumirlumir/npm-eslint-markdown/blob/main/packages/eslint-markdown/src/rules/no-tab.ts) rule in [eslint-markdown](https://github.com/lumirlumir/npm-eslint-markdown).

Currently, the `no-tab` rule checks the entire Markdown source for tab characters and replaces them with the number of spaces specified by `tabWidth` when applying autofixes. The `skipCode` and `skipInlineCode` options allow code blocks and inline code to be excluded from checks, but there are no options to exclude math regions.

As a result, tabs used for source alignment or indentation within math expressions are reported as errors and replaced with spaces when applying autofixes. Users who want to manage whitespace within math expressions separately from regular Markdown text cannot currently do so through the rule options.

`@eslint/markdown` supports math parsing through `languageOptions: { math: true }`. The resulting `math` and `inlineMath` nodes can be used to identify math regions and exclude them from checks. See the [official documentation](https://github.com/eslint/markdown#enabling-math-latex-in-both-commonmark-and-gfm) for configuration details.

Both options should default to `true`, consistent with the existing options for excluding code regions.

### 1. Add Option Types, Schema Entries, and Default Values

Add the following options to `packages/eslint-markdown/src/rules/no-tab.ts`.

| Option | Type | Default | Behavior |
| --- | --- | --- | --- |
| `skipMath` | `boolean` | `true` | Excludes tabs within math blocks from checks |
| `skipInlineMath` | `boolean` | `true` | Excludes tabs within inline math from checks |

Add descriptions and default values to `RuleOptions`, following the style of the existing options.

```ts
/**
 * `true` allows tabs in all math blocks.
 * @default true
 */
skipMath: boolean;
/**
 * `true` allows tabs in all inline math.
 * @default true
 */
skipInlineMath: boolean;
```

Register both options as `boolean` properties in `meta.schema`.

```ts
skipMath: {
  type: 'boolean',
},
skipInlineMath: {
  type: 'boolean',
},
```

Add the following default values to `meta.defaultOptions`.

```diff
 defaultOptions: [
   {
     skipCode: true,
     skipInlineCode: true,
+    skipMath: true,
+    skipInlineMath: true,
     tabWidth: 4,
   },
 ],
```

The two options must work independently. For example, with `skipMath: true` and `skipInlineMath: false`, math blocks should be excluded, while tabs within inline math should still be checked.

### 2. Register Math Node Ranges as Excluded Regions

Read the new options in `create()` and add `math` and `inlineMath` listeners using the same pattern as the existing `code` and `inlineCode` listeners.

```ts
const [{ skipCode, skipInlineCode, skipMath, skipInlineMath, tabWidth }] =
  context.options;
```

Example listener implementations:

```ts
math(node) {
  if (skipMath) skipRanges.push(sourceCode.getRange(node));
},

inlineMath(node) {
  if (skipInlineMath) skipRanges.push(sourceCode.getRange(node));
},
```

Reusing the existing `SkipRanges` utility and the checking logic in `'root:exit'` will skip both error reporting and autofixes within excluded math regions.

Use **the ranges of nodes that the parser actually recognizes as math**, rather than searching for math delimiters with a separate regular expression. When `languageOptions.math` is disabled, no math nodes are generated, so ordinary text that resembles math must remain subject to checks even when the new options are enabled.

### 3. Add Tests

Add the following scenarios to `packages/eslint-markdown/src/rules/no-tab.test.ts`.

- Verify that tabs within math blocks and inline math are allowed with the default rule options.
- Verify that the corresponding math regions are excluded when `skipMath: true` and `skipInlineMath: true` are explicitly specified.
- Verify that `skipMath: false` causes tabs within math blocks to be reported and autofixed according to `tabWidth`.
- Verify that `skipInlineMath: false` causes tabs within inline math to be reported and autofixed.
- Verify that math blocks and inline math are handled independently when the two options have different values.
- Verify that, when math and regular text appear together, tabs within math are excluded while tabs in regular text are still reported and fixed.
- Verify that, when math parsing is disabled, tabs within ordinary text resembling math are still checked even if both options are `true`.

Add the following configuration to tests that require math parsing.

```ts
languageOptions: {
  math: true,
},
```

For example, a test that allows tabs in math blocks while checking inline math and regular text can be written as follows.

```ts
{
  name: 'skips tabs in math blocks but fixes tabs in inline math and prose',
  languageOptions: {
    math: true,
  },
  options: [
    {
      skipMath: true,
      skipInlineMath: false,
      tabWidth: 2,
    },
  ],
  code: '$$\na\t=b\n$$\n\n$x\t=y$ and\ttext',
  output: '$$\na\t=b\n$$\n\n$x  =y$ and  text',
  errors: [
    { messageId: 'noTab' },
    { messageId: 'noTab' },
  ],
},
```

Give each test a name that clearly describes its scenario, and do not generate test cases using loops. Use the existing test helper to verify behavior in both CommonMark and GFM.

### 4. Update the Documentation

Update `website/docs/rules/no-tab.md` to include the following.

- Add `skipMath: true` and `skipInlineMath: true` to the option list and default configuration example.
- Explain each option’s type, default value, and applicable region.
- Add examples of correct and incorrect usage with the options enabled and disabled.
- State that `languageOptions: { math: true }` is required to exclude math regions.
- Explain that tabs within excluded math regions are not autofixed.

An example configuration is shown below. This example assumes that the `markdown` and `md` plugins have already been registered.

```js
{
  files: ['**/*.md'],
  language: 'markdown/gfm',
  languageOptions: {
    math: true,
  },
  rules: {
    'md/no-tab': ['error', {
      skipCode: true,
      skipInlineCode: true,
      skipMath: true,
      skipInlineMath: true,
      tabWidth: 4,
    }],
  },
}
```

Ensure that math parsing is also enabled for documentation examples that are checked automatically. Specifying rule options alone does not enable math parsing.

### 5. Validation

For a fresh clone, run `npm install` from the repository root and wait for installation to complete successfully before starting implementation or validation.

After making the changes, run the following commands from the repository root.

```sh
npm run test
npm run build
npm run lint
```

**Comment by khakaa** [comment]
Hi @lumirlumir, I'd like to work on this. Could you assign it to me?

**Comment by lumirlumir** [comment]
Thanks for your interest in this issue! A PR would be welcome.
